### DISCUSSION TITLE
Extending QueryGroupOptions

### DISCUSSION THREAD

**Discussion by gavol** [grafana/grafana#38254] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Enable custom data sources to extend the QueryGroupOptions structure that today is limited to few parameters.

**Why is this needed**:
With the proliferation of data sources, having a fixed number of options in QueryGroupOptions looks like a limitation. While parameters like maxDataPoints and cacheTimeout may be somehow universally useful, there may be others strictly related to a well-defined data source.

As a use case, we are developing a data source plugin and we are exploring how to pass some custom parameters that should be common for a graph and not for the single query (i.e., in the same way as maxDataPoints is used). That may include, for instance, different versions of a down-sampling algorithm.

Thanks in advance and apologies if I ignore that what I am asking for is already possible.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
